packages/eas-cli/src/utils/profiles.ts

```diff
@@ -57,11 +59,15 @@ async function readProfileAsync<T extends ProfileType>({
   profileName?: string;
 }): Promise<EasProfile<T>> {
   if (type === 'build') {
-    return (await EasJsonUtils.getBuildProfileAsync(
+    const buildProfile = await EasJsonUtils.getBuildProfileAsync(
       easJsonAccessor,
       platform,
       profileName
-    )) as EasProfile<T>;
+    );
+
+    maybePrintBuildProfileDeprecationWarnings(buildProfile, profileName);
```

Comment:
Maybe we should somehow check if this was already called. So we don't print the same warning multiple times when calling `readProfileAsync`.